app/scripts/modules/amazon/src/loadBalancer/configure/common/targetGroupValidators.spec.ts

```diff
@@ -0,0 +1,162 @@
+  healthyThreshold: 10,
+  name: 'targetgroup',
+  port: 7001,
+  protocol: 'HTTP',
+  targetType: 'ip',
+  unhealthyThreshold: 2,
+  account: 'test',
+  cloudProvider: 'aws',
+  healthTimeout: 1000,
+  healthInterval: 10,
+  loadBalancerNames: ['loadbalancer1', 'loadbalancer2'],
+  region: 'us-east-1',
+  type: '',
+};
+
+describe('Target Group validators', () => {
+  describe('of used names', () => {
+    it('returns an error if the name exists already', () => {
+      const existingGroups = {
+        test: {
+          'us-east-1': ['targetgroup'],
+        },
+      };
+      const actual = Validators.isNameInUse(existingGroups, 'test', 'us-east-1')(mockTargetGroup.name);
+      const expected = 'There is already a target group in test:us-east-1 with that name.';
+      expect(actual).toEqual(expected);
+    });
+
+    it('returns an null if it does not exist', () => {
+      const existingGroups = {
+        test: {
+          'us-east-1': ['targetgroup2'],
+        },
+      };
+      const actual = Validators.isNameInUse(existingGroups, 'test', 'us-east-1')(mockTargetGroup.name);
+      expect(actual).toEqual(null);
+    });
+  });
+
+  describe('of name length', () => {
+    it('returns error if the name is >32 additional characters', () => {
+      const actual = Validators.isNameLong(12)(`${mockTargetGroup.name}areallylongname`);
+      const expected =
+        'Target group name is automatically prefixed with the application name and cannot exceed 32 characters in length.';
+      expect(actual).toEqual(expected);
+    });
+
+    it('returns null if the name is < 32', () => {
+      const actual = Validators.isNameLong(12)(mockTargetGroup.name);
+      expect(actual).toEqual(null);
+    });
+  });
+
+  describe('of duplicate names across the load balancer', () => {
+    it('returns an error if the name exists', () => {
+      const actual = Validators.isDuplicateName(['targetgroup'])(mockTargetGroup.name);
+      const expected = 'Duplicate target group name in this load balancer.';
+      expect(actual).toEqual(expected);
+    });
+
+    it('returns null if the name is new', () => {
+      const actual = Validators.isDuplicateName(['targetGroup2'])(mockTargetGroup.name);
+      expect(actual).toEqual(null);
+    });
+  });
+
+  describe('of health check timeout constraints', () => {
+    const tg = {
+      ...mockTargetGroup,
+      protocol: 'TCP',
+      healthCheckProtocol: 'HTTP',
+    };
+
+    it('should have a 6s timeout', () => {
+      const actual = Validators.isValidTimeout(tg)('8');
+      const expected = 'HTTP health check timeouts for TCP/TLS target groups must be 6s';
+      expect(actual).toEqual(expected);
+    });
+
+    it('should be 6s and is valid', () => {
+      const actual = Validators.isValidTimeout(tg)('6');
+      expect(actual).toEqual(null);
+    });
+
+    it('should have a 10s timeout', () => {
+      const actual = Validators.isValidTimeout({ ...tg, healthCheckProtocol: 'HTTPS' })('9');
```

Comment:
i found this double-override (first in the definition of `tg` and then inline in the `it` blocks) a bit hard to follow. Thoughts on just defining multiple variables at the top with friendlier names? e.g.
```ts
    const httpTargetGroup = {
      ...mockTargetGroup,
      protocol: 'TCP',
      healthCheckProtocol: 'HTTP',
    };

    const httpsTargetGroup = {
      ...mockTargetGroup,
      protocol: 'TCP',
      healthCheckProtocol: 'HTTPS',
    };

```

Reply:
No strong opinions. I always lean toward repeating code as little as possible, but Chris expressed a similar concern as you so I will go make the references a bit more readable.